src/Paramol/SqlClient/TSqlDecimalValue.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace Paramol.SqlClient
+{
+    /// <summary>
+    ///     Represents a T-SQL DECIMAL parameter value.
+    /// </summary>
+    public class TSqlDecimalValue : IDbParameterValue
+    {
+        /// <summary>
+        ///     The single instance of this value.
+        /// </summary>
+        public static readonly TSqlDecimalValue Instance = new TSqlDecimalValue();
+
+        private readonly decimal _value;
+        private readonly byte _precision;
+        private readonly byte _scale;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public TSqlDecimalValue()
+        {
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="precision"></param>
+        /// <param name="scale"></param>
+        public TSqlDecimalValue(decimal value, byte precision = 18, byte scale = 0)
+        {
+            _value = value;
+
+            if (precision < 1 || precision > 38)
+                throw new ArgumentOutOfRangeException("precision", precision, "The precision must be between 1 and 38.");
+
+            _precision = precision;
+
+            if (scale > precision)
```

Comment:
You mentioned value objects about the scale and precision. Is this ok or would you like the rules encapsulated in value objects.